Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Extensions/InteractionSourceExtensions.cs

```diff
@@ -1,138 +1,49 @@
-                            {
-                                if (durationInSeconds.Equals(float.MaxValue))
-                                {
-                                    simpleHapticsController.SendHapticFeedback(hapticsFeedback, intensity);
-                                }
-                                else
-                                {
-                                    simpleHapticsController.SendHapticFeedbackForDuration(hapticsFeedback, intensity, TimeSpan.FromSeconds(durationInSeconds));
-                                }
-                                return;
-                            }
-                        }
-                    }
-                }
-            }, true);
-#elif UNITY_EDITOR_WIN
-            StartHaptics(interactionSource.id, intensity, durationInSeconds);
-#endif // !UNITY_EDITOR
-        }
-
-        public static void StopHaptics(this InteractionSource interactionSource)
-        {
-            if (!WindowsApiChecker.UniversalApiContractV4_IsAvailable && !Application.isEditor)
-            {
-                return;
-            }
-
-#if !UNITY_EDITOR
-            UnityEngine.WSA.Application.InvokeOnUIThread(() =>
-            {
-                IReadOnlyList<SpatialInteractionSourceState> sources = SpatialInteractionManager.GetForCurrentView().GetDetectedSourcesAtTimestamp(PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
-
-                foreach (SpatialInteractionSourceState sourceState in sources)
-                {
-                    if (sourceState.Source.Id.Equals(interactionSource.id))
-                    {
-                        sourceState.Source.Controller.SimpleHapticsController.StopFeedback();
-                    }
-                }
-            }, true);
-#elif UNITY_EDITOR_WIN
-            StopHaptics(interactionSource.id);
-#endif // !UNITY_EDITOR
-        }
-
-#if !UNITY_EDITOR
+#if WINDOWS_UWP
         public static IAsyncOperation<IRandomAccessStreamWithContentType> TryGetRenderableModelAsync(this InteractionSource interactionSource)
         {
             IAsyncOperation<IRandomAccessStreamWithContentType> returnValue = null;
 
             if (WindowsApiChecker.UniversalApiContractV5_IsAvailable)
             {
+                IReadOnlyList<SpatialInteractionSourceState> sources = null;
+
                 UnityEngine.WSA.Application.InvokeOnUIThread(() =>
                 {
-                    IReadOnlyList<SpatialInteractionSourceState> sources = SpatialInteractionManager.GetForCurrentView().GetDetectedSourcesAtTimestamp(PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
+                    sources = SpatialInteractionManager.GetForCurrentView()?.GetDetectedSourcesAtTimestamp(PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
+                }, true);
 
-                    for (var i = 0; i < sources.Count; i++)
+                for (var i = 0; i < sources?.Count; i++)
```

Comment:
Does i < null == false?

Reply:
That's an interesting point! I'll double check that this functions expectedly

Reply:
Under https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types#operators

>For the relational operators (<, >, <=, >=), if one or both operands are null, the result is false.